### PR TITLE
Ensure that public framework headers can be cleanly imported from outside the engine root.

### PR DESCRIPTION
**Pull request by chinmaygarde**
The new GN rules also ensure that deletions of exported headers correctly remove the stale entry from the set of exported headers. Previously, this required developers to test applications that consumed the framework (causing issues like https://github.com/flutter/engine/pull/4262).

### CODE REVIEW

File: sky/tools/install_framework_headers.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    # Ignore only "not found" errors.
+    if e.errno != errno.ENOENT:
+      raise e
+
+  # Create the directory to copy the files to.
+  if not os.path.isdir(args.location):
+    os.makedirs(args.location)
+
+  # Copy all files specified in the args.
+  for file in args.headers:
```
Comment (cbracken) [inline sky/tools/install_framework_headers.py]: Should we use a different name than `file` here? IIRC it's a Python builtin, which I guess we're shadowing here.

Comment (chinmaygarde) [reply sky/tools/install_framework_headers.py]: Done.

File: shell/platform/darwin/ios/framework/Source/FlutterUmbrellaImport.m [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2017 The Flutter Authors. All rights reserved.
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterUmbrellaImport.m]: `s/Flutter/Chromium/` for the time being according to @Hixie

File: sky/tools/install_framework_headers.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# Copyright 2017 The Flutter Authors. All rights reserved.
```
Comment (cbracken) [inline sky/tools/install_framework_headers.py]: `s/Flutter/Chromium/`

**Comment by cbracken** [comment]
LGTM (modulo two comments)
lgtm

**Comment by krisgiesing** [comment]
LGTM

**Pull request merged**